core/src/test/java/com/linecorp/armeria/server/ServerTest.java

```diff
@@ -513,6 +523,142 @@ void versionMetrics() {
+        // One gauge set for defaultVirtualHost and another for *.virtual.com
+        assertThat(validityGauges.size()).isEqualTo(2);
+        assertThat(daysValidityGauges.size()).isEqualTo(2);
+
+        assertThat(meterRegistry.find(CERT_VALIDITY_GAUGE_NAME)
+                                .tag("common_name", hostnamePattern)
+                                .tag("hostname", defaultHostName)
+                                .gauge().value()).isOne();
+        assertThat(meterRegistry.find(CERT_VALIDITY_DAYS_GAUGE_NAME)
+                                .tag("common_name", hostnamePattern)
+                                .tag("hostname", defaultHostName)
+                                .gauge().value()).isPositive();
+    }
+
+    @Test
+    void tlsMetricGivenCertificateChainNotExpired() {
+        final InputStream expiredCertificateChain = getClass().getResourceAsStream("certificate-chain.pem");
+        final InputStream pk = getClass().getResourceAsStream("pk.key");
+
+        final MeterRegistry meterRegistry = PrometheusMeterRegistries.newRegistry();
+        Server.builder()
+              .service("/", (ctx, req) -> HttpResponse.of(200))
+              .meterRegistry(meterRegistry)
+              .tls(expiredCertificateChain, pk)
+              .build();
+
+        assertThatGauge(meterRegistry, CERT_VALIDITY_GAUGE_NAME, "localhost").isOne();
+        assertThatGauge(meterRegistry, CERT_VALIDITY_DAYS_GAUGE_NAME, "localhost").isPositive();
+        assertThatGauge(meterRegistry, CERT_VALIDITY_GAUGE_NAME, "test.root.armeria").isOne();
+        assertThatGauge(meterRegistry, CERT_VALIDITY_DAYS_GAUGE_NAME, "test.root.armeria").isPositive();
+    }
+
+    @Test
+    void tlsMetricGivenExpired() throws CertificateException {
+        final Calendar calendar = Calendar.getInstance();
+        calendar.add(Calendar.DAY_OF_MONTH, -1);
+        final Date notAfter = calendar.getTime();
+        calendar.add(Calendar.DAY_OF_MONTH, -1);
+        final Date notBefore = calendar.getTime();
+        final SelfSignedCertificate ssc = new SelfSignedCertificate("localhost", notBefore, notAfter);
+
+        final MeterRegistry meterRegistry = PrometheusMeterRegistries.newRegistry();
+        Server.builder()
+              .service("/", (ctx, req) -> HttpResponse.of(200))
+              .meterRegistry(meterRegistry)
+              .tls(ssc.certificate(), ssc.privateKey())
+              .build();
+
+        assertThatGauge(meterRegistry, CERT_VALIDITY_GAUGE_NAME, "localhost").isZero();
+        assertThatGauge(meterRegistry, CERT_VALIDITY_DAYS_GAUGE_NAME, "localhost").isEqualTo(-1);
+    }
+
+    @Test
+    void tlsMetricGivenCertificateChainExpired() {
+        final InputStream expiredCertificateChain = getClass()
+                .getResourceAsStream("expired-certificate_chain.pem");
+        final InputStream pk = getClass().getResourceAsStream("expire_pk.key");
```

Comment:
Should we replace `_` with `-`?
```suggestion
                .getResourceAsStream("expired-certificate-chain.pem");
        final InputStream pk = getClass().getResourceAsStream("expire-pk.key");
```